engine-standalone-storage/src/relayer_db/types.rs

```diff
@@ -146,12 +186,41 @@ pub struct TransactionRow {
     pub output: Vec<u8>,
 }
 
+struct TransactionRowBlock(pub i64);
+
+impl From<TransactionRowBlock> for u64 {
+    fn from(value: TransactionRowBlock) -> Self {
+        // set negative values to 0
+        let res: u64 = value.0.max(0).try_into().unwrap();
```

Comment:
In the case of `value.0` is negative, you get an error.

Reply:
`unwrap_or_default` would not be ideal. Can lead to unexpected errors. Implementing `TryFrom` is better. `unwrap` anywhere in production code is unacceptable.

Reply:
Of course, it depends on the business logic. If a negative value is unacceptable, we should return an error here. In this case, you need to implement the `TryFrom` trait rather than just `From`.